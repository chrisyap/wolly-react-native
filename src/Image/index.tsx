import styled from 'styled-components/native';
import {
  border,
  BorderProps,
  color,
  ColorProps,
  layout,
  LayoutProps,
  position,
  PositionProps,
  space,
  SpaceProps,
} from 'styled-system';

export interface ImgProp extends BorderProps, ColorProps, LayoutProps, PositionProps, SpaceProps {}

const Img = styled.Image<ImgProp>`
  ${border};
  ${color};
  ${layout};
  ${position};
  ${space};
`;

export default Img;
