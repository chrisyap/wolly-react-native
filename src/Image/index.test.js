import { shallow } from 'enzyme';
import React from 'react';
import Image from './index';

describe('Image', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Image />);
    expect(wrapper).toMatchSnapshot();
  });
});
