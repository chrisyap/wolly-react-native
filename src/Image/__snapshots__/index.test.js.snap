// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<StyledNativeComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "getSize": [Function],
      "getSizeWithHeaders": [Function],
      "inlineStyle": InlineStyle {
        "rules": Array [
          "
  ",
          [Function],
          ";
  ",
          [Function],
          ";
  ",
          [Function],
          ";
  ",
          [Function],
          ";
  ",
          [Function],
          ";
",
        ],
      },
      "prefetch": [Function],
      "prefetchWithMetadata": [Function],
      "queryCache": [Function],
      "render": [Function],
      "resolveAssetSource": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "StyledNativeComponent",
      "target": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
